lib/gosu/compat.rb

```diff
@@ -0,0 +1,179 @@
+# This whole file is about backward compatibility
+
+# First - define some methods to tell the world about their old code
+
+module Gosu
+  DEPRECATION_STACKTRACE_LINES = 1
+
+  # Basically copy&paste from Gem but remove the date-part.
+  def self.deprecate(klass, name, repl)
+    klass.class_eval {
+      old = "_deprecated_#{name}"
+      alias_method old, name
+      define_method name do |*args, &block|
+        Gosu.deprecation_message(self, name, repl)
+        send old, *args, &block
+      end
+    }
+  end
+
+  def self.deprecate_const(name, repl)
+    send(:remove_const, name) if const_defined?(name)
+
+    @@_deprecated_constants ||= {}
+    @@_deprecated_constants[name] = repl
+  end
+
+  # The advantage of this approach is that the old Names won't show up in
+  # any Codecompletion (like IRB) or *.constants, so you have to know the
+  # exact old name to acutally use the old const.
+  # However define the value after the first call, to show the warning only once.
+  def self.const_missing(const_name)
+    if @@_deprecated_constants && repl = @@_deprecated_constants[const_name]
+      Gosu.deprecation_message(self, const_name, repl)
+      const_get(repl)
+    else
+      super
+    end
+  end
+
+  def self.deprecation_message(klass_or_full_message, name=nil, repl=nil)
+    @@_deprecations_shown ||= {}
+
+    msg = if klass_or_full_message.is_a?(String) && name.nil? && repl.nil?
```

Comment:
Gosu's own codebase uses `and` instead of `&&` in conditionals (same in `const_missing`). Maybe I'll change my mind in the future, but for now let's keep it consistent.